src/app/downloads/downloads.component.ts

```diff
@@ -123,74 +171,114 @@ export class DownloadsComponent implements OnInit, OnDestroy {
       endDate: endOfTodayUTC().toDate(),
       interval: INTERVAL_DAILY
     };
-    this.store.dispatch(new CastleFilterAction({filter: this.filter}));
+    if (newFilter.standardRange) {
+      this.filter.standardRange = newFilter.standardRange;
+    }
+    if (newFilter.range) {
+      this.filter.range = newFilter.range;
+    }
+    if (newFilter.beginDate) {
+      this.filter.beginDate = newFilter.beginDate;
+    }
+    if (newFilter.endDate) {
+      this.filter.endDate = newFilter.endDate;
+    }
+    if (newFilter.interval) {
+      this.filter.interval = newFilter.interval;
+    }
+    if (newFilter.podcastSeriesId) {
+      const routerParams = {
+        beginDate: this.filter.beginDate.toISOString(),
+        endDate: this.filter.endDate.toISOString(),
+        standardRange: this.filter.standardRange,
+        range: this.filter.range.join(',')
+      };
+      // if episodes are on the route, keep them on the route
+      // but do not put them on this.filter yet because we need to check for incoming changes
+      if (newFilter.episodeIds) {
+        routerParams['episodes'] = newFilter.episodeIds.join(',');
+      }
+      this.router.navigate([newFilter.podcastSeriesId, 'downloads', this.filter.interval.key, routerParams]);
+    } else {
+      this.store.dispatch(new CastleFilterAction({filter: this.filter}));
+    }
   }
 
   setDefaultEpisodeFilter() {
```

Comment:
some overlapping logic now in construction + nav to routerParams between this & the setDefaultFilter -- could make a helper ? not pressing

Reply:
true, there probably is a little bit more to be evolved and thought out here as far as cleaner code... I imagine it will continue coming along. Thanks for your kind attention! ❤️